cmd/krew/cmd/index.go

```diff
@@ -62,14 +66,68 @@ var indexAddCmd = &cobra.Command{
 	Example: "kubectl krew index add default " + constants.IndexURI,
 	Args:    cobra.ExactArgs(2),
 	RunE: func(_ *cobra.Command, args []string) error {
-		return indexoperations.AddIndex(paths.IndexBase(), args[0], args[1])
+		return indexoperations.AddIndex(paths, args[0], args[1])
 	},
 }
 
+var indexDeleteCmd = &cobra.Command{
+	Use:   "remove",
+	Short: "Remove a configured index",
+	Long: `Remove a configured index repository which is used to discover and
+install plugins from.
+
+If there are plugins installed from the specified index, the index cannot be
+removed (unless --force is used). Removing indexes while there are plugins
+installed from them is not supported behavior.`,
+	Args: cobra.ExactArgs(1),
+	RunE: indexDelete,
+}
+
+func indexDelete(_ *cobra.Command, args []string) error {
+	name := args[0]
+
+	pl, err := installation.InstalledPluginsFromIndex(paths.InstallReceiptsPath(), name)
+	if err != nil {
+		return errors.Wrap(err, "failed querying plugins installed from the index")
+	}
+	klog.V(4).Infof("Found %d plugins from index", len(pl))
+
+	if len(pl) > 0 && !*forceIndexDelete {
+		var names []string
+		for _, v := range pl {
+			names = append(names, v.Name)
+		}
+
+		internal.PrintWarning(os.Stderr, `Plugins [%s] are still installed from index %q.
+			"Removing indexes while there are plugins installed from them may result in
+			"unsupported behavior (use --force to ignore this check).`, strings.Join(names, ", "), name)
+		return errors.Errorf("there are still plugins installed from this index")
+	}
+
+	err = indexoperations.DeleteIndex(paths, name)
+	if os.IsNotExist(err) {
+		if *forceIndexDelete {
+			klog.V(4).Infof("Index not found, but --force is used, so not returning an error")
+			return nil // success if --force specified and index does not exist.
+		}
+		return errors.Errorf("index %q does not exist", name)
+	}
+	return errors.Wrap(err, "error while removing the plugin index")
+}
+
+var (
+	forceIndexDelete *bool
+)
```

Comment:
I would look for this var declaration right after the imports. Not sure if we have a recommended pattern for this, though..

Reply:
Done.